frappe/public/js/frappe/views/calendar/calendar.js

```diff
@@ -277,7 +277,12 @@ frappe.views.Calendar = Class.extend({
 						event[me.field_map.end] = me.get_system_datetime(moment(endDate).subtract(1, "s"));
 				}
 
-				frappe.set_route("Form", me.doctype, event.name);
+				// Set defaults from filters
+				cur_list.filter_area.get() // [[...], [...], [...]]
+					.filter(f => (f[2] == '=') && (f[0] == me.doctype)) // [[me.doctype, 'key', '=', 'value'], ...]
+					.forEach(filter => event[filter[1]] = filter[3]); // event['key'] = 'value'
+
+				frappe.ui.form.make_quick_entry(me.doctype, null, null, event);
```

Comment:
`list_view.js` has a `make_new_doc` method which does this. CalendarView extends from ListView, it would be better if we find a way to use that method.